Assets/Mirror/Runtime/NetworkWriter.cs

```diff
@@ -54,12 +57,29 @@ public void SetLength(long value)
 
         public void Write(string value)
         {
-            // BinaryWriter doesn't support null strings, so let's write an extra boolean for that
-            // (note: original HLAPI would write "" for null strings, but if a string is null on the server then it
-            //        should also be null on the client)
+            // write bool for null support first
+            // (note: original HLAPI would write "" for null strings, but if a
+            //        string is null on the server then it should also be null
+            //        on the client)
             writer.Write(value != null);
-            if (value != null) 
-                writer.Write(value);
+
+            // write string with same method as NetworkReader
+            if (value != null)
+            {
+                // check if within max size to avoid allocation attacks etc.
+                int numBytes = encoding.GetByteCount(value);
```

Comment:
this is the write,  not read,   allocation attacks not possible here.

GetByteCount(value)  is slow as it has to go through the entire string.

Instead,  call GetBytes as you do below and write convertedBytes